sample_consensus/include/pcl/sample_consensus/sac_model_ellipse3d.h

```diff
@@ -0,0 +1,307 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2012-, Open Perception, Inc.
+ *
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder(s) nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#pragma once
+
+#include <pcl/sample_consensus/sac_model.h>
+#include <pcl/sample_consensus/model_types.h>
+
+namespace pcl
+{
+  /** \brief SampleConsensusModelEllipse3D defines a model for 3D ellipse segmentation.
+    *
+    * The model coefficients are defined as:
+    *   - \b center.x : the X coordinate of the ellipse's center
+    *   - \b center.y : the Y coordinate of the ellipse's center
+    *   - \b center.z : the Z coordinate of the ellipse's center 
+    *   - \b semi_axis.u : semi-major axis length along the local u-axis of the ellipse
+    *   - \b semi_axis.v : semi-minor axis length along the local v-axis of the ellipse
+    *   - \b normal.x : the X coordinate of the normal's direction 
```

Comment:
You decided to use the normal and the semi-major axis in the coefficients. What would your thoughts be about using the semi-major and the semi-manor axes instead? For the circle in 3D, using the normal is the obvious choice, since there are no semi-major/minor axes. But for an ellipse, my intuition would be that the semi-axes have more meaning than the normal. And it seems more straightforward to use both semi-axes instead of normal and one semi-axis. What do you think?

Reply:
In the end, that is just a matter of interpretation. You are correct, but the approach of one normal and the semi-major axis is also correct. I decided to follow the same approach as used in the Circle3D because, if needed, one can directly retrieve the information regarding the plane where the Ellipse3D lies (in my application this was an obvious choice). Otherwise, the user still has to further compute the cross-product between the two semi-axes.